File: src/statusline/env.go
package main

import (
	"os"
	"path/filepath"
	"strings"
)

func envSegment(cwd, projectDir string) string {
	if cwd == "" {
		return ""
	}
	var parts []string

	home := os.Getenv("HOME")

	// Project root (only shown if cwd is inside a subdirectory)
	if projectDir != "" && projectDir != cwd {
		rootDisplay := filepath.Base(projectDir)
		if home != "" && strings.HasPrefix(projectDir, home) {
			rootDisplay = "~" + projectDir[len(home):]
		}
		parts = append(parts, CCyan+"\uf07c "+osc8Link(editorFileURL(projectDir), rootDisplay)+Rst)
	}

	// Directory with ~ shortening and editor deep link
	short := cwd
	if home != "" && strings.HasPrefix(cwd, home) {
		rest := cwd[len(home):]
		if rest == "" {
			short = "~/"
		} else {
			short = "~" + rest
		}
	}
	p := strings.Split(short, "/")
	if len(p) > 3 {
		short = p[0] + "/" + p[1] + "/.../" + p[len(p)-1]
	}
	parts = append(parts, CCyan+"\ue5ff "+osc8Link(editorFileURL(cwd), short)+Rst)

	return strings.Join(parts, Sep)
}

File: dot-claude/hooks/statusline/env.js
// Environment segment — project root + directory with editor deep links

const os = require('os');
const path = require('path');
const { RST, SEP, osc8Link, editorFileURL } = require('./shared');

function envSegment(cwd, root) {
  if (!cwd) return '';
  const parts = [];

  // Project root (only shown if cwd is inside a subdirectory)
  if (root && root !== cwd) {
    const name = path.basename(root);
    parts.push('\x1b[38;2;97;175;239m\uf07c ' + osc8Link(editorFileURL(root), name) + RST);
  }

  // Directory with ~ shortening and editor deep link
  const home = os.homedir();
  let short = cwd;
  if (home && cwd.startsWith(home)) {
    short = '~' + (cwd.slice(home.length) || '/');
  }
  const p = short.split('/');
  if (p.length > 3) {
    short = p[0] + '/' + p[1] + '/.../' + p[p.length - 1];
  }
  parts.push('\x1b[38;2;97;175;239m\ue5ff ' + osc8Link(editorFileURL(cwd), short) + RST);

  return parts.join(SEP);
}

module.exports = { envSegment };
